src/js/stores/ui.js

```diff
@@ -117,6 +119,15 @@ define(function (require, exports, module) {
          * @type {number}
          */
         _toolbarWidth: null,
+        
+        /**
+         * A list of document IDs that are centered after panel resized. If a document's id is not within the list,
+         * it will be centered when it is selected.
+         *
+         * @private
+         * @type {Array.<number>}
+         */
+        _centeredDocumentIDs: null,
```

Comment:
Why not use a `Set` here instead of an Array? That way you can do constant-time membership checks, additions and removals. See here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set